src/celeritas/phys/PrimaryGenerator.hh

```diff
@@ -0,0 +1,67 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2020-2022 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/phys/PrimaryGenerator.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include <vector>
+
+#include "ParticleParams.hh"
+#include "Primary.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Primary generator construction arguments.
+ */
+struct PrimaryGeneratorOptions
+{
+    int       pdg;      //!< Primary PDG number
+    real_type energy;   //!< [MeV]
+    Real3     position; //!< [cm]
+    Real3     direction;
+    size_type num_events;
+    size_type primaries_per_event;
+
+    //! Whether the options are valid
+    explicit operator bool() const
+    {
+        return pdg != 0 && energy >= 0 && num_events > 0
+               && primaries_per_event > 0;
+    }
```

Comment:
Without initializers, these scalar types are uninitialized; in particular for the object `LDemoArgs::primary_gen_options`. If you're unlucky (and on my machine, I seem to be always unlucky), this makes `operator bool()` return `true` and it's not possible to load HepMC3 files (unless an explicit `primary_gen_options` initializes the fields to zero? not tested).

Reply:
Ugh, you're right: and since there's no unit test for this class it doesn't run through valgrind. I'll add a test and push a fix.

Reply:
Fixed by #472